crates/common/tedge_config/src/tedge_config_cli/new.rs

```diff
@@ -83,6 +84,28 @@ impl TEdgeConfig {
 
         Ok(mqtt_config)
     }
+
+    pub fn mqtt_client_auth_configs(
```

Comment:
Under which conditions this function returns an error is not very clear.

1. It looks to me like the function doesn't throw errors in any case (there's no `?` or `return Err(..)`), so the function should just return a thing instead of result.
2. There is a comment (that I remember having written) saying that if either option is not set, we return an error. Right now, it looks like we don't return an error, so this is either a regression or the comment should be removed.
3. As more of a general note, if a function is public, and even more so if it returns a result, it should be documented, at least to tell when it's supposed to fail.

Reply:
Yes, the error is useless., removed it.